docs/OSUpgrades.md

```diff
@@ -14,83 +14,98 @@ These bootimages are built using [coreos-assembler](https://github.com/coreos/co
-node. As part of that `pivot.service` runs (which is
-`Before=kubelet.service`, so before the cluster starts). It detects the
-reference to the updated `machine-os-content` and pulls it, uses it to upgrade, and
-reboots.
+When that succeeds, the above process of `machine-config-daemon-firstboot.service`
+runs which extracts OS updates from the `machine-os-content` container image,
+and the control plane nodes each reboot (before `kubelet.service` has started).
 
-The master machines come up and form a cluster.
+When the control plane nodes reboot and form a cluster, the bootstrap
+node is torn down.
 
-# Workers and management via the MCO
+# Workers
 
 At this point, Ignition has been executed, and that only runs once.
+The `machine-config-daemon-firstboot.service` is no longer used for OS updates.
 
 The master machines use the [machine-api-operator](https://github.com/openshift/machine-api-operator) to
 boot the workers.  Each worker pulls Ignition configs from the MCS running
-on the control plane, which has the exact same pivot code.  They also
-upgrade/reboot, and then join the cluster.
+on the control plane.  The exact same process of performing an upgrade
+and reboot happens for each worker.
 
-Thereafter, the MCO takes over fully. The `machine-config-daemon` daemonset
-lands on the master nodes, and will start watching for updates to
-the `machine-os-content` from the release image, as well as any changes
-to the `machineconfig` objects.
+# Management via the Machine Config Daemon
+
+After a node (whether control plane or worker) has joined the cluster, the MCO
+takes over.  Previously, each individual node was running systemd units;
+now changes are coordinated via the operator.
+
+When the administrator starts an `oc adm upgrade`, if a new `machine-os-content`
+is provided in the release image, it will be rolled out to the control plane
+and workers.
 
 Every change now will be managed by a `machineconfigpool`, ensuring
 that only 1 machine at a time is changed (via `maxUnavailable: 1` default).
 
-However, because of the early pivot, the master nodes have already booted
-the desired version (and the Ignition config generated at bootstrap time should
-match the in-cluster one) so the MCO has nothing to do.
+# MCD host upgrade execution
 
-But now the MCO is fully in control of operating system updates.
-The next time the admin does an `oc adm upgrade`, if a new `machine-os-content`
-is provided in the release image, it will be rolled out to the masters
-and workers.
+Today mostly because of [SELinux reasons](https://bugzilla.redhat.com/show_bug.cgi?id=1839065) the
+MCD copies itself to the host, and runs as part of the host context.
+Then it provides updated content to `rpm-ostreed.service`, which is already
+a daemon running on the host.
+
+The MCD tries to watch the systemd journal for relevant service and proxy logs,
+so you *should* be able to `oc -n openshift-machine-config-operator logs pod/machine-config-daemon-...`
```

Comment:
should we specify `oc -n openshift-machine-config-operator logs -c machine-config-daemon pod/machine-config-daemon-...` for clarity

Reply:
agree